zbar/zbar.py

```diff
@@ -181,4 +182,32 @@ def scan(self, image):
             symbols.append(Symbol(sym_name, sym_data, sym_quality, sym_loc))
             symbol = _ZB.zbar_symbol_next(symbol)
         assert len(symbols) == num_symbols
-        return symbols
\ No newline at end of file
+        return symbols
+
+    def rgb2gray(self,rgb):
+        '''
+            converts rgb to grayscale image
+            rgb is of type numpy.ndarray
+        '''
+        return numpy.dot(rgb[...,:3], [0.299, 0.587, 0.114])
+
+    def scan_from_image(self,image_filename):
```

Comment:
As in the comment to `setup.py`, this is probably better as an example than a core part of the library, if only to avoid the scipy dependency.